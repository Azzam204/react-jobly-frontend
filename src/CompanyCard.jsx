import React from "react";
import { Link } from "react-router-dom";
import './CompanyCard.css'

/** Show company's basic info
 * 
 * rendered for each company by CompanyList */ 

function CompanyCard({comp}) {
  return (
    <div>
      <Link className="CompanyCard card" to={`/companies/${comp.handle}`}>
        <div className="card-body">
          <h6 className="card-title">
            {comp.name}
            <img className="float-end ml-5" src={`https://dummyimage.com/500x300/000/fff&text=${comp.name}`} alt={comp.name} />
          </h6>
          <p><small>{comp.description}</small></p>
        </div>
      </Link>
    </div>
  )
}

export default CompanyCard