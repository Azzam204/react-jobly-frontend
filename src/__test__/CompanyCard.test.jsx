import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import CompanyCard from "../CompanyCard";

const comp = {
  name: 'company',
  handle: 'comp',
  description: 'comp description'
}

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <CompanyCard comp={comp} />
    </MemoryRouter>
  )
})

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <CompanyCard comp={comp} />
    </MemoryRouter>
  )

  expect(asFragment()).toMatchSnapshot();
})