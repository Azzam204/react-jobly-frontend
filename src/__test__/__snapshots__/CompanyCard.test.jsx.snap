// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <a
      class="CompanyCard card"
      href="/companies/comp"
    >
      <div
        class="card-body"
      >
        <h6
          class="card-title"
        >
          company
          <img
            alt="company"
            class="float-end ml-5"
            src="https://dummyimage.com/500x300/000/fff&text=company"
          />
        </h6>
        <p>
          <small>
            comp description
          </small>
        </p>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
